core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -330,7 +330,7 @@ private[spark] object JettyUtils extends Logging {
 
           // redirect the HTTP requests to HTTPS port
           httpConnector.setName(REDIRECT_CONNECTOR_NAME)
-          collection.addHandler(createRedirectHttpsHandler(securePort, scheme))
+          collection.addHandler(createRedirectHttpsHandler(connector, scheme))
```

Comment:
@vanzin , AFAIK in this code, at that time when we create a redirect handler, the https `ServerConnector` hasn't yet started, so we couldn't get the real port from it, it will return -1 as we call `getLocalPort`.